R/epGRanges.R

```diff
@@ -0,0 +1,46 @@
+## Equipartition GRanges:
+## Splits a GRanges object into a specified number of parts with equal (total) width.
+## Each part is a GRanges and all parts are returned as GRangesList object. 
+
+## Alexandre Kuhn (alexandre.m.kuhn@gmail.com)
+
+epGRanges<-function(gr,partitionNb) {
+        if (partitionNb<1) stop("The number of partition should be at least 1.")
+        else if (partitionNb==1) return(GRangesList(gr))
+        else
+        {
+        grLength<-length(gr)
+        cs<-cumsum(width(gr))
+
+        partitionWidth<-floor(cs[grLength]/partitionNb) #last partition is longer
+        #determine partitions' upper bounds (ub)
+        upperbound<-seq(partitionWidth,cs[grLength],partitionWidth)
+        upperbound[partitionNb]<-cs[grLength]
+
+        #for each ub, identify the gr element to which it locates
+        ubIndex<-sapply(upperbound,function(x){which(cs>=x)[1]})
```

Comment:
`findInterval()` is a more efficient way of doing this. But I think what you really want to do from here is form a new Partitioning using the widths of the input ranges with `PartitioningByWidth()`. Then, use `disjoin()` to break them up. If you pass `with.revmap=TRUE` to `disjoin()` you can recover the original seqnames and starts of the ranges.